src/parser/shared/modules/items/GemChecker.tsx

```diff
@@ -0,0 +1,288 @@
+import { Trans } from '@lingui/react/macro';
+import { Gem as GemItem } from 'common/ITEMS/Item';
+import { ItemLink } from 'interface';
+import Analyzer from 'parser/core/Analyzer';
+import { Item, Gem as EventGem } from 'parser/core/Events';
+import { ItemHelper } from 'parser/core/itemHelper';
+import { GemBoxRowEntry } from 'interface/guide/components/Preparation/GemSubSection/GemBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import { gemById } from 'common/ITEMS/thewarwithin/gems';
+import GEAR_SLOTS from 'game/GEAR_SLOTS';
+
+class GemChecker extends Analyzer {
+  //Which slots can have a gem slots added
+  static twoAddableGemSlots = [GEAR_SLOTS.NECK, GEAR_SLOTS.FINGER1, GEAR_SLOTS.FINGER2];
+  static oneAddableGemSlot = [GEAR_SLOTS.HEAD, GEAR_SLOTS.WAIST, GEAR_SLOTS.WRISTS];
+
+  get GemableSlots(): Record<number, JSX.Element> {
+    return {}; //Keeping form with EnchantChecker and having the user override this
+  }
+
+  get GemableGear(): any {
```

Comment:
You shouldn't use the `any` type.
```suggestion
  get GemableGear(): Record<number, Item> {
```

Reply:
Cool, I was mimicking what I saw from enchanted trying to keep the coding style.  I love strong typing so no problem there.